### PR TITLE
add terminology troubleshooting log

### PR DESCRIPTION
**Pull request by fil512**

### CODE REVIEW

File: hapi-fhir-base/src/main/java/ca/uhn/fhir/context/support/IValidationSupport.java [hunk 1]
```diff
@@ -990,6 +995,15 @@ public IIdType getResourceId() {
...
 			return myReverse;
 		}
+
+		@Override
+		public String toString() {
+			return new ToStringBuilder(this)
+					.append("sourceValueSetUrl", mySourceValueSetUrl)
+					.append("targetSystemUrl", myTargetSystemUrl)
+					.append("targetValueSetUrl", myTargetValueSetUrl)
+					.toString();
```
Comment (jmarchionatto) [inline hapi-fhir-base/src/main/java/ca/uhn/fhir/context/support/IValidationSupport.java:1005]: myReverse could be important

File: hapi-fhir-validation/src/main/java/org/hl7/fhir/common/hapi/validation/support/ValidationSupportChain.java [hunk 2]
```diff
@@ -66,14 +69,25 @@ public TranslateConceptResults translateConcept(TranslateCodeRequest theRequest)
...
 				}
 
-				retVal.getResults().addAll(translations.getResults());
+				if (!translations.isEmpty()) {
+					if (ourLog.isDebugEnabled()) {
+						ourLog.debug(
+								"{} found {} concept translation{} for {}",
+								next.getName(),
+								translations.size(),
+								translations.size() > 1 ? "s" : "",
```
Comment (jmarchionatto) [inline hapi-fhir-validation/src/main/java/org/hl7/fhir/common/hapi/validation/support/ValidationSupportChain.java:78]: Clever!

File: hapi-fhir-validation/src/main/java/org/hl7/fhir/common/hapi/validation/support/ValidationSupportChain.java [hunk 3]
```diff
@@ -84,6 +98,9 @@ public boolean isValueSetSupported(ValidationSupportContext theValidationSupport
 		for (IValidationSupport next : myChain) {
 			boolean retVal = next.isValueSetSupported(theValidationSupportContext, theValueSetUrl);
 			if (retVal) {
+				if (ourLog.isDebugEnabled()) {
+					ourLog.debug("ValueSet {} found in {}", theValueSetUrl, next.getName());
```
Comment (jmarchionatto) [inline hapi-fhir-validation/src/main/java/org/hl7/fhir/common/hapi/validation/support/ValidationSupportChain.java:102]: thought: may be worth using lambdas to save the previous extra check: 
ourLog.debug("ValueSet {} found in {}", () -> theValueSetUrl, () -> next.getName());

**Review by jmarchionatto — APPROVED** [review]
Very useful addition! Approved with a few comments

File: hapi-fhir-validation/src/main/java/org/hl7/fhir/common/hapi/validation/support/ValidationSupportChain.java (hunk 3 above)
Comment (fil512) [reply hapi-fhir-validation/src/main/java/org/hl7/fhir/common/hapi/validation/support/ValidationSupportChain.java:102]: Oh right I forgot you could do that.  I'll leave it this way for now for consistency.

**Pull request merged**